transifex/common/utils.py

```diff
@@ -130,6 +159,12 @@ def _consume_rule(string):
             <<< ('other', '{OTHER}')
     """
 
+    def _get_rule_num(rule):
```

Comment:
Should we put something like this in `common`?

```python
class _Rules(object):
    """ Usage:

            >>> Rules = _Rules()

            >>> Rules[0].string
            <<< "zero"

            >>> Rules[1].integer
            <<< 1

            >>> Rules['two'].string
            <<< "two"

            >>> Rules['few'].integer
            <<< 3
    """

    Rule = collections.namedtuple("Rule", [integer, string])

    def __getitem__(self, index):
        rules = (self.Rule(0, "zero"), self.Rule(1, "one"),
                 self.Rule(2, "two"), self.Rule(3, "few"),
                 self.Rule(4, "many"), self.Rule(5, "other"))

        for rule in rules:
            if index in rule:
                return rule

        raise KeyError(index)


Rules = _Rules()

```

We can even use it in txc. Lots of tedious code over there doing the same thing.

Reply:
_Slightly_ overengineered alternative https://gist.github.com/kbairak/5ba23227abf5edfa33354724c52af8a7

Reply:
Happy to do this, but I'd like to separate it from this PR to merge it soon.